src/qwed_new/core/cache.py

```
@@ -315,159 +475,182 @@ def clear(self, pattern: str = None) -> int:
+    # Redis path -- mode determines failure semantics
+    if _redis_cache is None or _redis_cache.tenant_id != tenant_id or _redis_cache.mode != mode:
+        _redis_cache = RedisCache(tenant_id=tenant_id, mode=mode)
+
+    return _redis_cache
 
 
 def cached_verify(
     dsl_code: str,
     variables: Optional[list] = None,
-    verify_fn: callable = None
+    verify_fn: callable = None,
```

Ruff (0.15.14)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)